src/control/controlpushbutton.h

```diff
@@ -1,69 +1,42 @@
-/***************************************************************************
-                          controlpushbutton.h  -  description
-                             -------------------
-    begin                : Wed Feb 20 2002
-    copyright            : (C) 2002 by Tue and Ken Haste Andersen
-    email                :
- ***************************************************************************/
-
-/***************************************************************************
- *                                                                         *
- *   This program is free software; you can redistribute it and/or modify  *
- *   it under the terms of the GNU General Public License as published by  *
- *   the Free Software Foundation; either version 2 of the License, or     *
- *   (at your option) any later version.                                   *
- *                                                                         *
- ***************************************************************************/
-
-#ifndef CONTROLPUSHBUTTON_H
-#define CONTROLPUSHBUTTON_H
+#pragma once
 
+#include "control/controlbuttonmode.h"
 #include "control/controlobject.h"
 
-/**
-  *@author Tue and Ken Haste Andersen
-  */
-
 class ControlPushButton : public ControlObject {
     Q_OBJECT
   public:
-    enum ButtonMode {
-         PUSH = 0,
-         TOGGLE,
-         POWERWINDOW,
-         LONGPRESSLATCHING,
-         TRIGGER,
-    };
-
-    static QString buttonModeToString(int mode) {
+    static QString buttonModeToString(ControlButtonMode mode) {
         switch(mode) {
-            case ControlPushButton::PUSH:
-                return "PUSH";
-            case ControlPushButton::TOGGLE:
-                return "TOGGLE";
-            case ControlPushButton::POWERWINDOW:
-                return "POWERWINDOW";
-            case ControlPushButton::LONGPRESSLATCHING:
-                return "LONGPRESSLATCHING";
-            case ControlPushButton::TRIGGER:
-                return "TRIGGER";
-            default:
-                return "UNKNOWN";
+        case ControlButtonMode::Push:
+            return QStringLiteral("Push");
+        case ControlButtonMode::Toggle:
+            return QStringLiteral("Toggle");
+        case ControlButtonMode::PowerWindow:
+            return QStringLiteral("PowerWindow");
+        case ControlButtonMode::LongPressLatching:
+            return QStringLiteral("LongPressLatching");
+        case ControlButtonMode::Trigger:
+            return QStringLiteral("Trigger");
+        default:
+            // Without the default branch we get compiler warnings
+            DEBUG_ASSERT(false);
+            return "Unknown";
         }
```

Comment:
I assume you are referring to "control reaches end of non-void function [-Wreturn-type]"? If you just do this: we'll still get `-Wswitch` (which we want), but not `-Wreturn-type`.
```suggestion
        }
        DEBUG_ASSERT(false);
        return QStringLiteral("Unknown");
```